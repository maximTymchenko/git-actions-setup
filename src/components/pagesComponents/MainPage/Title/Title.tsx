import React, { FC } from 'react';
import Button from '../../../common/Button';
import triangleRight from '../../../../assets/img/triangle-right.svg';

const Title: FC = () => {
    return (
        <div data-testid="title" className="main-title__body">
            <div className="main-title__wrapper">
                <div className="main-title__container">
                    <h2 className="main-title">
                        Your Superior Team <span className="main-title__span">to</span>
                    </h2>
                </div>
                <div className="main-title__animation">
                    <div className="main-title__image">
                        <div className="main-title__icon" />
                        <div className="main-title__border" />
                    </div>
                </div>
            </div>

            <div className="main-title__desktop-button">
                <Button type={'watch'} className="button" desc={'Watch now'} icon={triangleRight}>
                    Watch now
                </Button>
            </div>

            <div className="main-title__mobile-button">
                <Button type="circle" className="button" desc={'Watch now'} icon={triangleRight}>
                    Watch now
                </Button>
            </div>
        </div>
    );
};

export default Title;
