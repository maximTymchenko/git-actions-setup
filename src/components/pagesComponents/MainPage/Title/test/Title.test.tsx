import React from 'react';
import Title from '../Title';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(
        <Title
            title="Shall we chat quickly?"
            text="Even a 5-minute intro call could potentially save you countless hours of endless
                    back-and-forth emails."
        />,
    );
    expect(getByTestId('title')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(
            <Title
                title="Shall we chat quickly?"
                text="Even a 5-minute intro call could potentially save you countless hours of endless
                    back-and-forth emails."
            />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
