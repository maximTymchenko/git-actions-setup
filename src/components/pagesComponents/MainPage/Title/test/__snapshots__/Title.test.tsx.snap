// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="main-title__body"
  data-testid="title"
>
  <div
    className="main-title__wrapper"
  >
    <div
      className="main-title__container"
    >
      <h2
        className="main-title"
      >
        Your Superior Team 
        <span
          className="main-title__span"
        >
          to
        </span>
      </h2>
    </div>
    <div
      className="main-title__animation"
    >
      <div
        className="main-title__image"
      >
        <div
          className="main-title__icon"
        />
        <div
          className="main-title__border"
        />
        <div
          className="main-title__mobile-button"
        >
          <button
            className="button button--circle button--undefined button--undefined"
            data-testid="button"
          >
            Watch now
            <img
              alt="Watch now"
              className="button__icon button__icon--undefined"
              data-testid="button__icon"
              src="test-file-stub"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="main-title__desktop-button"
  >
    <button
      className="button button--watch button--undefined button--undefined"
      data-testid="button"
    >
      Watch now
      <img
        alt="Watch now"
        className="button__icon button__icon--undefined"
        data-testid="button__icon"
        src="test-file-stub"
      />
    </button>
  </div>
</div>
`;
